crates/noirc_frontend/src/ast/mod.rs

```diff
@@ -14,14 +14,49 @@ pub use structure::*;
 
 use crate::{token::IntType, util::vecmap, IsConst};
 
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum UnresolvedArraySize {
+    Variable,
+    Fixed(u64),
+    FixedVariable(Ident),
+}
+
+impl UnresolvedArraySize {
+    pub fn is_fixed(&self) -> bool {
+        matches!(self, UnresolvedArraySize::Fixed(_))
+    }
+
+    pub fn is_fixed_variable(&self) -> bool {
+        matches!(self, UnresolvedArraySize::FixedVariable(_))
+    }
+
+    pub fn is_variable(&self) -> bool {
+        !self.is_fixed()
+    }
+
+    pub fn is_subtype_of(&self, argument: &UnresolvedArraySize) -> bool {
+        (self.is_fixed() && argument.is_variable()) || (self == argument)
```

Comment:
This, and the whole impl I believe, can be removed since they are unused. `is_subtype_of` should only be needed during type checking, which should occur on Types rather than UnresolvedTypes.

Reply:
Good catch ya I think I left it from when it was just `ArraySize`